common/lib/core-interfaces/src/provider.ts

```diff
@@ -25,8 +25,11 @@
  export type FluidObjectProviderKeys<T, TProp extends keyof T = keyof T> =
     string extends TProp ? never : number extends TProp? never : // exclude indexers [key:string |number]: any
     TProp extends keyof Exclude<T[TProp], undefined> // TProp is a property of T, and T[TProp]
-        ? TProp
-        :never;
+        ? TProp extends
```

Comment:
My suggestion was to do `T[TProp] extends Required< Required<T>[TProp] >[TProp]`, to make sure the type itself matches, not just the keys.

Your way accepts this (event more contrived!) example but mine doesn't:  `z: { z: { z: boolean } };`

PS - I think using `Required` is slightly more readable since it doesn't need the other type param